src/modules/dashboard/components/ColonyHome/ColonyMembers/MembersSubsection.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useCallback, useMemo } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import Maybe from 'graphql/tsutils/Maybe';
+
+import NavLink from '~core/NavLink';
+import Heading from '~core/Heading';
+import { Tooltip } from '~core/Popover';
+import Icon from '~core/Icon';
+
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import useAvatarDisplayCounter from '~utils/hooks/useAvatarDisplayCounter';
+import { Colony, useLoggedInUser, BannedUser, UserProfile } from '~data/index';
+import { COLONY_TOTAL_BALANCE_DOMAIN_ID } from '~constants';
+import { Address } from '~types/index';
+import { useTransformer } from '~utils/hooks';
+import { getAllUserRoles } from '~modules/transformers';
+import { hasRoot, canAdminister } from '~modules/users/checks';
+
+import styles from './ColonyMembers.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.ColonyHome.ColonyMembers.MembersSubsection.title',
+    defaultMessage: `{isContributors, select,
+      true { Contributors }
+      other { Watchers }
+    } {hasCounter, select,
+        true { ({count})}
+        false {}
+      }`,
+  },
+  loadingData: {
+    id: 'dashboard.ColonyHome.ColonyMembers.MembersSubsection.loadingData',
+    defaultMessage: 'Loading members information...',
+  },
+  reputationFetchFailed: {
+    id: `dashboard.ColonyHome.ColonyMembers.MembersSubsection.reputationFetchFailed`,
+    defaultMessage: `Failed to fetch the colony's {isContributors, select,
+      true { contributors }
+      other { watchers }
+    }`,
+  },
+  tooltipText: {
+    id: 'dashboard.ColonyHome.ColonyMembers.MembersSubsection.tooltipText',
+    defaultMessage: `{isContributors, select,
+      true {Contributors are members of the Colony who have earned reputation.}
+      other { Watchers are members of the Colony
+         who currently don’t have reputation. }
+    }`,
+  },
+});
+
+type BannedMember = Maybe<
+  Pick<BannedUser, 'id' | 'eventId' | 'banned'> & {
+    profile?:
+      | Maybe<Pick<UserProfile, 'displayName' | 'username' | 'walletAddress'>>
+      | undefined;
+  }
+>;
+
+interface Props {
+  colony: Colony;
+  currentDomainId?: number;
+  maxAvatars?: number;
+  members: Maybe<string[]>;
+  bannedMembers?: BannedMember[];
+  isContributors: boolean;
+}
+
+const UserAvatar = HookedUserAvatar({ fetchUser: true });
+
+const displayName = 'dashboard.ColonyHome.ColonyMembers.MembersSubsection';
+
+const MembersSubsection = ({
+  colony: { colonyName },
+  members,
+  bannedMembers,
+  isContributors,
+  colony,
+  currentDomainId = COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  maxAvatars = 12,
```

Comment:
Maybe extract the constant out somewhere or set as a default since it is used twice?

Reply:
Yeah, the default value `12` should be a `const`, but I like the approach of having the number of avatars configurable via props